core/trino-main/src/main/java/io/trino/operator/WorkProcessorUtils.java

```diff
@@ -352,8 +352,7 @@ else if (processor.isBlocked()) {
     {
         @Nullable
         WorkProcessor.Process<T> process;
-        @Nullable
-        ProcessState<T> state;
+        ProcessState<T> state = ProcessState.yield();
```

Comment:
Add a comment why this is a reasonable initial state, as it's not obvious.

Also, did you consider introducing a dedicated INITIAL state type?